tests/integration/components/frost-text-test.js

```diff
@@ -16,29 +19,318 @@ describeComponent(
+      const spellcheck = 'true'
+
+      this.set('spellcheck', spellcheck)
+      this.render(hbs`
+        {{frost-text
+          spellcheck=spellcheck
+        }}
+      `)
+
+      expect(
+        this.$('input').attr('spellcheck'),
+        'spellcheck attribute is set'
+      ).to.eql('true')
+    })
+
+    it('sets error class', function () {
+      const error = 'error'
+
+      this.set('error', error)
+
+      this.render(hbs`
+        {{frost-text
+          class=error
+        }}
+      `)
+
+      expect(
+        this.$('.frost-text').hasClass('error'),
+        'error class is set'
+      ).to.be.true
     })
 
     it('only renders the clear icon in insert', function () {
-      this.set('external', 'temp')
-      this.render(hbs`{{frost-text value=external}}`)
-      expect(this.$('.frost-text-clear')).to.have.length(1)
+      this.render(hbs`
+        {{frost-text}}
+      `)
+
+      run(() => this.$('input').val('Test').trigger('input'))
+
+      expect(
+        this.$('.frost-text').hasClass('is-clear-visible'),
+        'class "is-clear-visible" is set'
+      ).to.be.true
 
-      this.set('external', 'change')
-      expect(this.$('.frost-text-clear')).to.have.length(1)
+      expect(
+        this.$('.frost-text').hasClass('is-clear-enabled'),
+        'class "is-clear-enabled" is set'
+      ).to.be.true
+    })
+
+    it('text cleared on button click', function () {
+      this.render(hbs`
+        {{frost-text
+          value="Test"
+        }}
+      `)
+
+      this.$('input').focus().val('').trigger('input')
+
+      expect(
+        this.$('input').val()
+      ).to.eql('')
     })
 
     it('hook attr grabs frost-text as expected', function () {
-      this.render(hbs`{{frost-text hook='my-text'}}`)
+      this.render(hbs`
+          {{frost-text 
+            hook='my-text'
+          }}
+      `)
+
+      expect(
+        $hook('my-text-input').hasClass('frost-text-input'),
+        'input hook is set'
+      ).to.be.true
+
+      expect(
+        $hook('my-text-clear').hasClass('frost-text-clear'),
+        'clear hook is set'
+      ).to.be.true
+    })
+
+    it('calls onKeyUp closure action', function () {
+      const externalActionSpy = sinon.spy()
+
+      this.on('externalAction', externalActionSpy)
+
+      this.render(hbs`
+        {{frost-text
+          onKeyUp=(action 'externalAction')
+        }}
+      `)
+
+      this.$('input').trigger('keyup')
+
+      expect(
+        externalActionSpy.called,
+        'onKeyUp closure action called'
+      ).to.be.true
+    })
+
+    it('calls onInput closure action', function () {
+      const externalActionSpy = sinon.spy()
+      const testValue = 'Test'
+
+      this.set('value', testValue)
+      this.on('externalAction', externalActionSpy)
+
+      this.render(hbs`
+        {{frost-text
+          onInput=(action "externalAction")
+          value=value
+        }}
+      `)
+
+      this.$('input').trigger('input')
+
+      const testObj = {
```

Comment:
Why create the testObj if you're going to just refer to the individual properties?